### ISSUE TITLE
Immutable settings

### ISSUE THREAD

**Issue by vincentarelbundock** [etiennebacher/altdoc#67]
It would be nice to host `sidebar.yml` and friends in a separate folder or in the root folder. We could then copy it over (with modifications and populating sections) to `docs/`.

The benefit would be that users could just wipe `docs/` and call `use_docsify()`, and have all of their hand-customized settings would remain intact.

We could add these settings to `.Rbuildignore` automatically.

**Comment by etiennebacher** [comment]
Yes, that's a bit related to #40

**Comment by vincentarelbundock** [comment]
Cool. This is definitely backward-breaking. So I'll think of a design and propose something eventually.

**Comment by vincentarelbundock** [comment]
Yeah, and I think the `man/` and `vignettes/` structure is very standard, so it would be good to replicate that exactly in `docs/` too.

That seems like a good investment for long term.

**Comment by vincentarelbundock** [comment]
Here's one user-interface idea.

Desiderata:

* Auto-population of sidebar is possible
* User can override it manually

Design:

* `use_*()` functions still creates a dummy file like now.
* This file includes entries like `$ALTDOC_NEWS` and `$ALTDOC_MAN`.
* The auto-population script looks for these entries
  - If present and `NEWS` is available, then it auto-populates
  - If present and `NEWS` is unavailable, then it removes the `altdoc` marker from the dummy file
  - If absent, then it autopopulates with our own structure.

That way, if a user wants to only show the docs for a subset of functions, all they have to do is remove `$ALTDOC_MAN` from the immutable settings file, and write in their own structure.

Thoughts?

**Comment by etiennebacher** [comment]
I like this, but I wonder whether `$ALTDOC_NEWS` for instance will work correctly in the `index.html` file for docute.

> That way, if a user wants to only show the docs for a subset of functions

But should this be possible? I think currently `pkgdown` errors if some missing functions are exported but not in `_pkgdown.yml`. Of course we don't have to mimic them 100%, but I think in this case this is a good thing to impose on users

**Comment by vincentarelbundock** [comment]
Ok, I'll play around with some variations of this and see if I can make it work in all three output formats.

> But should this be possible? 

One benefit is that devs can pick the order. Alphabetical is not the best order in my packages, where groups of functions go together conceptually.

But the main issue is that i have a LOT of exported and documented functions with `@keywords internal`. These methods have to be available to users otherwise `marginaleffects` breaks; they have to be documented because it is good practice; but they are not directly user-facing. Therefore, they should not appear in the man PDF because it would completely drown out the other core functions and make it unreadable.

We can't just look for `roxygen2` tags because package who don't use roxygen2 have a lot of different ways to achieve the same result. 

Allowing users full control is the easiest way out.

**Comment by etiennebacher** [comment]
> But the main issue is that i have a LOT of exported and documented functions with @keywords internal. These methods have to be available to users otherwise marginaleffects breaks;

That sounds weird, why do you have to export them?

**Comment by vincentarelbundock** [comment]
Yes, exactly. It's nice to have it exported for those who are working interactively to support new models, plus it can actually be quite convenient.

In any case, that is immaterial because it "costs" us nothing with the new design.

I've been working on this a lot over the weekend and am close to something I can show you. I'm pretty excited. It's very cool ;)
